accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+
+```xml
+<Project>
+  <ItemGroup Condition=" ('$(_IncludeWPFGlobs)' == 'true') ">
+    
+    <ApplicationDefinition Include="App.xaml"
+                           Condition="'$(EnableDefaultApplicationDefinition)' != 'false' And
+                                      Exists('$(MSBuildProjectDirectory)/App.xaml') And '$(MSBuildProjectExtension)' == '.csproj'"
+                           Generator="MSBuild:Compile" />
+
+    <ApplicationDefinition Include="Application.xaml"
+                           Condition="'$(EnableDefaultApplicationDefinition)' != 'false' And
+                                       Exists('$(MSBuildProjectDirectory)/Application.xaml') And '$(MSBuildProjectExtension)' == '.vbproj'"
+                           Generator="MSBuild:Compile" />
+
+
+    <Page Include="**/*.xaml"
+          Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(ApplicationDefinition)"
+          Condition="'$(EnableDefaultPageItems)' != 'false'"
+          Generator="MSBuild:Compile" />
+
+    <None Remove="**/*.xaml"
+          Condition="'$(EnableDefaultApplicationDefinition)' != 'false' And '$(EnableDefaultPageItems)' != 'false'" />
+  </ItemGroup>
+</Project>
+```
+
+The `AutoImport.props` files should keep their top-level conditions as simple as possible, so the .targets for a workload should set a property that the `AutoImport.props` file uses for its condition:
+
+```xml
+<PropertyGroup Condition=" ('$(TargetPlatformIdentifier)' == 'windows') And
+                         ('$(EnableDefaultItems)' == 'true') And ('$(UseWPF)' == 'true') And 
+                         ('$(_TargetFrameworkVersionValue)' != '$(_UndefinedTargetFrameworkVersion)') And 
+                         ('$(_TargetFrameworkVersionValue)' >= '$(_WindowsDesktopSdkTargetFrameworkVersionFloor)')">
+
+  <_IncludeWPFGlobs>true</_IncludeWPFGlobs>
+</PropertyGroup>
+```
+
+## Implementation
+
+### MSBuild and resolver features
+
+We will add the following capabilities to MSBuild [SdkResolvers](https://github.com/microsoft/msbuild/blob/master/src/Framework/Sdk/SdkResolver.cs):
+
+- Return any number of SDK paths (zero, one, or many)
+- Return MSBuild items and properties to add to the evaluation result
+
+### New resolver behavior
+
+We will create an MSBuild SDK resolver to handle workloads.  Its behavior should be the following:
+
+- Look up the Name of the requested SDK in the workload manifests, matching to the workload pack IDs
+- If a matching workload pack is found in a workload manifest, then get its version number, and use that version number to generate the path where that workload pack would be installed.  This should be the following, or similar: `<DOTNET ROOT>/packs/<PACK ID>/<PACK VERSION>/Sdk`
+- If the workload pack is installed (ie the path (except the final `Sdk`) exists), then return the generated path
+- If the workload pack is not installed, then the returned `SdkResult` should:
+  - Have `Success` set to true
```

Comment:
When would `Success` be false?

Reply:
It would be set to false if there was some sort of error (for example a malformed manifest or something).

Reply:
K. Might be good to call that out somewhere. I read "false" is for exceptional cases, while the case you call out is an expected case.